GitUI/UserControls/FilterToolBar.cs

```diff
@@ -305,6 +282,31 @@ private void UpdateBranchFilterItems()
                 await this.SwitchToMainThreadAsync();
                 BindBranches(branches);
             }).FileAndForget();
+
+            return;
+
+            void BindBranches(string[] branches)
+            {
+                var autoCompleteList = tscboBranchFilter.AutoCompleteCustomSource.Cast<string>();
+                if (!autoCompleteList.SequenceEqual(branches))
+                {
+                    tscboBranchFilter.AutoCompleteCustomSource.Clear();
+                    tscboBranchFilter.AutoCompleteCustomSource.AddRange(branches);
+                }
+
+                string filter = tscboBranchFilter.Items.Count > 0 ? tscboBranchFilter.Text : string.Empty;
+                string[] matches = branches.Where(branch => branch.IndexOf(filter, StringComparison.InvariantCultureIgnoreCase) >= 0).ToArray();
+
+                if (matches.Length == 0)
```

Comment:
```suggestion
                if (!matches.Any())
```

Reply:
Linq is not optimized a well as native code (even if I really hope it makes no difference here) so it is avoided, also suggested by vs. for me length is easier to read.

Reply:
In .NET Framework linq sufered from perf issues, in .NET some (most?) perf issues got addressed. I personally prefer to avoid linq where possible.